api/v0/ingest/schema/envelope.go

```diff
@@ -0,0 +1,101 @@
+package ingestion
+
+import (
+	"bytes"
+	"errors"
+
+	cidlink "github.com/ipld/go-ipld-prime/linking/cid"
+	crypto "github.com/libp2p/go-libp2p-core/crypto"
+	"github.com/libp2p/go-libp2p-core/record"
+	mh "github.com/multiformats/go-multihash"
+)
+
+const (
+	adSignatureCodec  = "/indexer/ingest/adSginature"
+	adSignatureDomain = "indexer"
+)
+
+type advSignatureRecord struct {
+	domain *string
+	codec  []byte
```

Comment:
are these for the smart/composable record format?

Reply:
They aren't, this comes from `go-libp2p-core/envelope`. @gammazero suggested using libp2p envelopes for signatures. It's the first time I've used them, so I may be using them wrong.